src/routes/auth/guards/auth.guard.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Inject,
+  Injectable,
+} from '@nestjs/common';
+import { IAuthRepository } from '@/domain/auth/auth.repository.interface';
+import { AuthService } from '@/routes/auth/auth.service';
+import { SiweMessageSchema } from '@/domain/auth/entities/siwe-message.entity';
+
+/**
+ * The AuthGuard should be used to protect routes that require authentication.
+ *
+ * It checks for the presence of a valid JWT access token in the request and
+ * verifies its validity before allowing access to the route.
+ *
+ * 1. Check for the presence of an access token in the request.
+ * 2. Verify the token's validity and decode it.
+ * 3. Ensure the decoded token is that of a valid SiweMessage.
+ * 4. Ensure the token is not expired or not yet valid.
```

Comment:
`4. Ensure the token is not expired and is already valid.`

(Maybe I'm mixing up things)

Reply:
You're spot on. Updated in 0d1fae29290f4892c659b2006c70c41e727c5623.